osu.Game/Tests/Visual/Multiplayer/TestMultiplayerClient.cs

```diff
@@ -89,13 +98,28 @@ public void ChangeUserBeatmapAvailability(int userId, BeatmapAvailability newBea
 
         protected override Task<MultiplayerRoom> JoinRoom(long roomId)
         {
-            var user = new MultiplayerRoomUser(api.LocalUser.Value.Id) { User = api.LocalUser.Value };
+            Debug.Assert(apiRoom != null);
 
-            var room = new MultiplayerRoom(roomId);
-            room.Users.Add(user);
+            var user = new MultiplayerRoomUser(api.LocalUser.Value.Id)
+            {
+                User = api.LocalUser.Value
+            };
 
-            if (room.Users.Count == 1)
-                room.Host = user;
+            var room = new MultiplayerRoom(roomId)
+            {
+                Settings =
+                {
+                    Name = apiRoom.Name.Value,
+                    BeatmapID = apiRoom.Playlist.Last().BeatmapID,
+                    RulesetID = apiRoom.Playlist.Last().RulesetID,
+                    BeatmapChecksum = apiRoom.Playlist.Last().Beatmap.Value.MD5Hash,
+                    RequiredMods = apiRoom.Playlist.Last().RequiredMods.Select(m => new APIMod(m)).ToArray(),
+                    AllowedMods = apiRoom.Playlist.Last().AllowedMods.Select(m => new APIMod(m)).ToArray(),
+                    PlaylistItemId = apiRoom.Playlist.Last().ID
```

Comment:
All multiplayer tests that join a room at start are failing on this (correctly) due to the playlist being empty. Honestly I'm not quite sure how it ever even worked in the first place. I think it didn't and the beatmap was shoved in at a later stage...?